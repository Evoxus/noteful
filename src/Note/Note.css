.Note {
  border: 1px solid black;
  height: 80px;
  width: 95%;
  max-width: 300px;
  margin: 5px;
  padding: 3px;
}

.Note h3 {
  font-size: 1.2em;
  margin-bottom: 15px;
}

.flexContainer {
  display: flex;
  justify-content: space-between;
  font-size: 0.85em;
}

.DeleteNote {
  border-radius: 5px;
  border: 1px solid black;
  background-color: #0b1f3a;
  color: white;
  height: 25px;
  width: 70px;
  cursor: pointer;
}

.DeleteNote:hover {
  background-color: #970000;
  color: white;
}

.NoteTitle {
  color: #bdbdbd;
  text-decoration: none;
  transition-duration: 100ms;
}

.NoteTitle:hover {
  transition-duration: 100ms;
  color: #646024;
}