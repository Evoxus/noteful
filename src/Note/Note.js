import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { format, parseISO } from 'date-fns';
import NoteContext from '../NoteContext';
import PropType from 'prop-types';
import './Note.css';

export default class Note extends Component {
  static defaultProps = {
    onDeleteNote: () => {},
  }

  static contextType = NoteContext;

  handleClickDelete = e => {
    e.preventDefault()
    const noteId = this.props.id
    fetch(`http://localhost:9090/notes/${noteId}`, {
      method: 'DELETE',
      headers: {
        'content-type': 'application/json'
      },
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(e => Promise.reject(e))
        return res.json()
      })
      .then(() => {
        this.props.onDeleteNote(noteId)
        this.context.deleteNote(noteId)
        // allow parent to perform extra behaviour
        
      })
      .catch(error => {
        console.error({ error })
      })
  }

  render() {
    return (
      <div className='Note'>
        <h3>
          <Link className='NoteTitle' to={`/note/${this.props.id}`} >{this.props.name}</Link>
        </h3>
        <p className='flexContainer'>
          <span className='Date'>
            Modified {this.props.modified && format(parseISO(this.props.modified), 'do MMM yyyy') }
          </span>
          <button className='DeleteNote' type='button' onClick={this.handleClickDelete}>DeleteNote</button>
        </p>
      </div>
    )
  }
}

Note.defaultProps = {
  note: {
    id: 1,
    name: 'Note',
    modified: new Date(),
    handleClickDelete: () => {},
  }
}

Note.propType = {
  name: PropType.string.isRequired,
  modified: PropType.string.isRequired,
  onDeleteNote: PropType.func.isRequired
}