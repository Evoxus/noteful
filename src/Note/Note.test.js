import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import Note from './Note';

describe('App component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter> <Note /> </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  })
  it('snapshot test', () => {
    const wrapper = shallow(<Note />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})