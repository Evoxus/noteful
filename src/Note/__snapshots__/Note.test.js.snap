// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component snapshot test 1`] = `
<div
  className="Note"
>
  <h3>
    <Link
      className="NoteTitle"
      to="/note/undefined"
    />
  </h3>
  <p
    className="flexContainer"
  >
    <span
      className="Date"
    >
      Modified 
    </span>
    <button
      className="DeleteNote"
      onClick={[Function]}
      type="button"
    >
      DeleteNote
    </button>
  </p>
</div>
`;
